src/lib.rs

```diff
@@ -24,10 +24,12 @@
 #![cfg_attr(not(debug_assertions), deny(clippy::used_underscore_binding))]
 
 pub mod anlz;
+pub mod device;
 pub mod pdb;
 pub mod setting;
 pub mod util;
 pub(crate) mod xor;
```

Comment:
Do we want to make the more internal modules private when they have a nicer API counterpart.

Reply:
Not sure. Maybe we should group the low-level stuff in a separate submodule or crate but leave it public for people who want to experiment or squeeze out maximum performance?